src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/blob.py

```diff
@@ -158,40 +208,29 @@ def _upload_blob(file_path, blob_name, blob_content_settings):
             'timeout': timeout
         }
 
+        if supported_api_version(ResourceType.DATA_STORAGE, min_api='2017-04-17') and tier:
+            create_blob_args['premium_page_blob_tier'] = tier
+
         if supported_api_version(ResourceType.DATA_STORAGE, min_api='2016-05-31'):
             create_blob_args['validate_content'] = validate_content
 
         return client.create_blob_from_path(**create_blob_args)
 
-    def _create_return_result(blob_name, blob_content_settings, upload_result=None):
-        return {
-            'Blob': client.make_blob_url(destination_container_name, blob_name),
-            'Type': blob_content_settings.content_type,
-            'Last Modified': upload_result.last_modified if upload_result else None,
-            'eTag': upload_result.etag if upload_result else None}
+    type_func = {
+        'append': upload_append_blob,
+        'block': upload_block_blob,
+        'page': upload_block_blob  # same implementation
+    }
+    return type_func[blob_type]()
 
-    upload_action = _upload_blob if blob_type == 'block' or blob_type == 'page' else _append_blob
-    logger = get_az_logger(__name__)
-    settings_class = get_sdk(ResourceType.DATA_STORAGE, 'blob.models#ContentSettings')
 
-    results = []
-    if dryrun:
-        logger.info('upload action: from %s to %s', source, destination)
-        logger.info('    pattern %s', pattern)
-        logger.info('  container %s', destination_container_name)
-        logger.info('       type %s', blob_type)
-        logger.info('      total %d', len(source_files))
-        results = []
-        for src, dst in source_files or []:
-            results.append(_create_return_result(dst, guess_content_type(src, content_settings, settings_class)))
-    else:
-        for src, dst in source_files or []:
-            logger.warning('uploading {}'.format(src))
-            guessed_content_settings = guess_content_type(src, content_settings, settings_class)
-            results.append(
-                _create_return_result(dst, guessed_content_settings, upload_action(src, dst, guessed_content_settings)))
+def _update_progress(current, total):
+    HOOK = APPLICATION.get_progress_controller(True)
```

Comment:
1. it's not global variable, lower case.
2. use named parameter to improve the readability.